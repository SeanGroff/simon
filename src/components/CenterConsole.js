// @flow
import React from 'react'
import styled from 'styled-components'
import Counter from '../components/Counter'
import StartStrict from '../containers/StartStrictContainer'
import OnOffSwitch from '../components/OnOffSwitch'

type Props = {
  power: boolean,
  counter: number,
  strictMode: boolean,
}

type BoxModelProps = {
  hasMargin?: boolean,
  hasPadding?: boolean,
}

const Wrapper = styled.div`
  position: absolute;
  border-radius: 100%;
  top: 50%;
  left: 50%;
  background-color: #ece7ee;
  border: 12px solid #333333;
  width: 220px;
  height: 220px;
  margin: -120px 0 0 -120px;
`

const TopHalf = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #333333;
  width: 220px;
  height: 110px;
  border-top-left-radius: 220px;
  border-top-right-radius: 220px;
  margin-top: -13px;
`

const BottomHalf = TopHalf.extend`
  flex-direction: column;
  background-color: #ece7ee;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 220px;
  border-bottom-right-radius: 220px;
  margin-top: 0;
`

const Header = styled.span`
  color: #ece7ee;
  font-family: 'Anton', sans-serif;
  font-size: 52px;
  letter-spacing: 2px;
  margin-top: 35px;
  text-transform: capitalize;
`

const ButtonsRow = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
`

const ButtonWrapper = styled.span`
  display: flex;
  flex-direction: column;
  margin: 5px;
`

const Label = styled.label`
  font-size: 14px;
  color: #222222;
  margin: ${({ hasMargin }: BoxModelProps): string =>
    hasMargin ? 'auto 5px 0 5px' : '0'};
  padding: ${({ hasPadding }: BoxModelProps): string =>
    hasPadding ? '0 0 5px 0' : '0'};
  text-transform: capitalize;
  text-align: center;
`

const StrictLight = styled.span`
  align-self: center;
  background: ${({ strict }) => (strict ? 'springgreen' : '#222222')};
  border: 2px solid #222222;
  border-radius: 100%;
  height: 8px;
  margin-top: 10px;
  width: 8px;
  transition: background 0.2s linear;
`

export default function CenterConsole({ power, counter, strictMode }: Props) {
  return (
    <Wrapper>
      <TopHalf>
        <Header>{'simon'}</Header>
      </TopHalf>
      <BottomHalf>
        <ButtonsRow>
          <ButtonWrapper>
            <Label hasPadding>{'count'}</Label>
            <Counter />
          </ButtonWrapper>
          <ButtonWrapper>
            <Label hasPadding>{'start'}</Label>
            <StartStrict startButton disabled={!power || counter} />
          </ButtonWrapper>
          <ButtonWrapper>
            <Label hasPadding>{'strict'}</Label>
            <StartStrict startButton={false} disabled={!power} />
          </ButtonWrapper>
          <StrictLight strict={strictMode} />
        </ButtonsRow>
        <ButtonsRow>
          <Label hasMargin>{'off'}</Label>
          <OnOffSwitch />
          <Label hasMargin>{'on'}</Label>
        </ButtonsRow>
      </BottomHalf>
    </Wrapper>
  )
}
