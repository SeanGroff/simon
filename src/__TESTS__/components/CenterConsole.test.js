import React from 'react';
import { shallow } from 'enzyme';
import CenterConsole from '../../components/CenterConsole';

it('renders correctly', () => {
  const component = shallow(<CenterConsole />);
  expect(component).toMatchSnapshot();
});
