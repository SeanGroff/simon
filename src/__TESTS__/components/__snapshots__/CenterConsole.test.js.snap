// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
        <styled.span>
            simon
        </styled.span>
    </styled.div>
    <styled.div>
        <styled.div>
            <styled.span>
                <styled.label
                    hasPadding={true}
                >
                    count
                </styled.label>
                <Counter
                    count={0}
                    on={false}
                />
            </styled.span>
            <styled.span>
                <styled.label
                    hasPadding={true}
                >
                    start
                </styled.label>
                <StartStrictButton
                    startButton={true}
                />
            </styled.span>
            <styled.span>
                <styled.label
                    hasPadding={true}
                >
                    strict
                </styled.label>
                <StartStrictButton
                    startButton={false}
                />
            </styled.span>
        </styled.div>
        <styled.div>
            <styled.label
                hasMargin={true}
            >
                off
            </styled.label>
            <OnOffSwitch
                powerOnOff={undefined}
            />
            <styled.label
                hasMargin={true}
            >
                on
            </styled.label>
        </styled.div>
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
            <styled.span>
                  simon
            </styled.span>
      </styled.div>
      <styled.div>
            <styled.div>
                  <styled.span>
                        <styled.label
                              hasPadding={true}
                        >
                              count
                        </styled.label>
                        <Counter
                              count={0}
                              on={false}
                        />
                  </styled.span>
                  <styled.span>
                        <styled.label
                              hasPadding={true}
                        >
                              start
                        </styled.label>
                        <StartStrictButton
                              startButton={true}
                        />
                  </styled.span>
                  <styled.span>
                        <styled.label
                              hasPadding={true}
                        >
                              strict
                        </styled.label>
                        <StartStrictButton
                              startButton={false}
                        />
                  </styled.span>
            </styled.div>
            <styled.div>
                  <styled.label
                        hasMargin={true}
                  >
                        off
                  </styled.label>
                  <OnOffSwitch
                        powerOnOff={undefined}
                  />
                  <styled.label
                        hasMargin={true}
                  >
                        on
                  </styled.label>
            </styled.div>
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
                    <styled.span>
                              simon
                    </styled.span>
          </styled.div>
          <styled.div>
                    <styled.div>
                              <styled.span>
                                        <styled.label
                                                  hasPadding={true}
                                        >
                                                  count
                                        </styled.label>
                                        <Counter
                                                  count={0}
                                                  on={false}
                                        />
                              </styled.span>
                              <styled.span>
                                        <styled.label
                                                  hasPadding={true}
                                        >
                                                  start
                                        </styled.label>
                                        <StartStrictButton
                                                  startButton={true}
                                        />
                              </styled.span>
                              <styled.span>
                                        <styled.label
                                                  hasPadding={true}
                                        >
                                                  strict
                                        </styled.label>
                                        <StartStrictButton
                                                  startButton={false}
                                        />
                              </styled.span>
                    </styled.div>
                    <styled.div>
                              <styled.label
                                        hasMargin={true}
                              >
                                        off
                              </styled.label>
                              <OnOffSwitch
                                        powerOnOff={undefined}
                              />
                              <styled.label
                                        hasMargin={true}
                              >
                                        on
                              </styled.label>
                    </styled.div>
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
                    <styled.span>
                              simon
                    </styled.span>
          </styled.div>
          <styled.div>
                    <styled.div>
                              <styled.span>
                                        <styled.label
                                                  hasPadding={true}
                                        >
                                                  count
                                        </styled.label>
                                        <Counter
                                                  count={0}
                                                  on={false}
                                        />
                              </styled.span>
                              <styled.span>
                                        <styled.label
                                                  hasPadding={true}
                                        >
                                                  start
                                        </styled.label>
                                        <StartStrictButton
                                                  startButton={true}
                                        />
                              </styled.span>
                              <styled.span>
                                        <styled.label
                                                  hasPadding={true}
                                        >
                                                  strict
                                        </styled.label>
                                        <StartStrictButton
                                                  startButton={false}
                                        />
                              </styled.span>
                    </styled.div>
                    <styled.div>
                              <styled.label
                                        hasMargin={true}
                              >
                                        off
                              </styled.label>
                              <OnOffSwitch
                                        powerOnOff={undefined}
                              />
                              <styled.label
                                        hasMargin={true}
                              >
                                        on
                              </styled.label>
                    </styled.div>
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
